libs/ledger-live-common/src/api/socket.ts

```diff
@@ -184,16 +185,25 @@ export const createDeviceSocket = (
                 total: data.length,
               });
 
-            notify(0);
-
-            for (let i = 0; i < data.length; i++) {
-              const r = await transport.exchange(Buffer.from(data[i], "hex"));
-              if (unsubscribed) return;
-              const status = r.readUInt16BE(r.length - 2);
-              if (status !== StatusCodes.OK) {
-                throw new TransportStatusError(status);
-              }
-              notify(i + 1);
+            // we use a promise to wait for the bulk to finish
+            await new Promise((resolve, reject) => {
+              let i = 0;
+              notify(0);
+              // we also use a subscription to be able to cancel the bulk if the user unsubscribes
+              bulkSubscription = transport.exchangeBulk(
+                data.map((d) => Buffer.from(d, "hex")),
+                {
+                  next: () => {
+                    notify(++i);
+                  },
+                  error: (e) => reject(e),
+                  complete: () => resolve(null),
```

Comment:
Should we reset `bulkSubscription` on `complete` to avoid an unnecessary call to `unsubscribe` ?

Reply:
we can, but it should always be a "no op". it's part of the remaining thing to do I think anyway because right now, unsubscribe isn't propagated all the way down the ipc